mountpoint-s3/src/upload.rs

```diff
@@ -72,9 +72,12 @@ impl<Client: ObjectClient> UploadRequest<Client> {
         bucket: &str,
         key: &str,
     ) -> ObjectClientResult<Self, PutObjectError, Client::ClientError> {
-        let params = PutObjectParams::new()
-            .trailing_checksums(true)
-            .storage_class(inner.storage_class.to_owned().unwrap());
+        let mut params = PutObjectParams::new().trailing_checksums(true);
+
+        if let Some(storage_class) = inner.storage_class.to_owned() {
+            params = params.storage_class(storage_class);
+        }
```

Comment:
nit: my preference would be for something like this, where `if let` just takes a reference and `clone` is called when passing the argument
```suggestion
        let mut params = PutObjectParams::new().trailing_checksums(true);
        if let Some(storage_class) = inner.storage_class {
            params.storage_class(storage_class.clone());
        }
```